### PR TITLE
Use custom data helper and legacycode copied class

### LINKED ISSUE #61: Inkompatible with 5.60+ ?
```
➜  org.project60.membership (new_master) grep -r -E
 "(CRM_Core_OptionGroup::getValue|CRM_Contact_BAO_Contact::contactTrashRestore|CRM_Contact_BAO_Contact::getPhoneDetails|CRM_Core_DAO::checkFieldExists|CRM_Core_DAO::createTempTableName|CRM_Core_OptionGroup::getLabel|_civicrm_api3_field_names|CRM_Core_BAO_Location::deleteLocationBlocks|_ipn_process_transaction)" .
 
./api/v3/Membership/Extend.php:  $contribution_status_completed = (int) CRM_Core_OptionGroup::getValue('contribution_status', 'Completed', 'name', 'String', 'value');
./CRM/Membership/TokenLogic.php:    $temp_contact2membership = CRM_Core_DAO::createTempTableName('p60m_token');
./CRM/Membership/CustomData.php:            $activity_type = CRM_Core_OptionGroup::getValue('activity_type', $activity_type, 'name');
```

### PR DESCRIPTION
**Pull request by jofranz**
to avoid dependencies

Closes #61

### CODE REVIEW

**Review by bjendres — APPROVED** [review]
Looks good to me.

Nitpick: next time please create two separate PRs, as it's two independent topics.

File: info.xml [hunk 1]
```diff
@@ -15,7 +15,6 @@
     <ver>5.38</ver>
   </compatibility>
   <requires>
-    <ext>legacycode</ext>
   </requires>
```
Comment (jofranz) [inline info.xml:18]: @bjendres Learning: An empty `requires` causes a crash on PHP 8.1. 
https://github.com/Project60/org.project60.membership/pull/68/files

_internal ref: 19731#note-29_

**Comment by bjendres** [comment]
>  An empty requires causes a crash on PHP 8.1.

WTF?

**Comment by jofranz** [comment]
> > An empty requires causes a crash on PHP 8.1.
> 
> WTF?

https://lab.civicrm.org/dev/core/-/issues/4707

**Pull request merged**